composer/datasets/ade20k.py

```diff
@@ -0,0 +1,331 @@
+
+        if self.base_size <= 0:
+            raise ValueError("base_size cannot be zero or negative.")
+
+        if self.min_resize_scale <= 0:
+            raise ValueError("min_resize_scale cannot be zero or negative")
+
+        if self.max_resize_scale < self.min_resize_scale:
+            raise ValueError("max_resize_scale cannot be less than min_resize_scale")
+
+    def initialize_object(self, batch_size, dataloader_hparams) -> DataloaderSpec:
+        self.validate()
+
+        if self.use_synthetic:
+            if self.split == 'train':
+                total_dataset_size = 20_206
+            elif self.split == 'val':
+                total_dataset_size = 2_000
+            else:
+                total_dataset_size = 3_352
+
+            dataset = SyntheticBatchPairDataset(
+                total_dataset_size=total_dataset_size,
+                data_shape=[3, self.final_size, self.final_size],
+                label_shape=[self.final_size, self.final_size],
+                num_classes=150,
+                num_unique_samples_to_create=self.synthetic_num_unique_samples,
+                device=self.synthetic_device,
+                memory_format=self.synthetic_memory_format,
+            )
+            collate_fn = None
+            device_transform_fn = None
+
+        else:
+            # Define data transformations based on data split
+            if self.split == 'train':
+                both_transforms = torch.nn.Sequential(
+                    RandomResizePair(min_scale=self.min_resize_scale,
+                                     max_scale=self.max_resize_scale,
+                                     base_size=(self.base_size, self.base_size)),
+                    RandomCropPair(crop_size=(self.final_size, self.final_size)),
+                    RandomHFlipPair(),
+                )
+                image_transforms = torch.nn.Sequential(
+                    PhotometricDistoration(brightness=32. / 255, contrast=0.5, saturation=0.5, hue=18. / 255),
+                    PadToSize(size=(self.final_size, self.final_size),
+                              fill=(int(0.485 * 255), int(0.456 * 255), int(0.406 * 255))))
+
+                target_transforms = PadToSize(size=(self.final_size, self.final_size), fill=0)
+            else:
+                both_transforms = None
+                image_transforms = transforms.Resize(size=(self.final_size, self.final_size),
+                                                     interpolation=TF.InterpolationMode.BILINEAR)
+                target_transforms = transforms.Resize(size=(self.final_size, self.final_size),
+                                                      interpolation=TF.InterpolationMode.NEAREST)
+            collate_fn = pil_image_collate
+            device_transform_fn = NormalizationFn(self.ignore_background)
+
+            dataset = ADE20k(
+                datadir=self.datadir,  # type: ignore
```

Comment:
```suggestion
                datadir=self.datadir,
```

Reply:
Added a check before to avoid type ignore